### PR TITLE
Add ACP Client with support for claude code, codex etc.

### PR DESCRIPTION
**Pull request by jasonkneen**
This pull request introduces support for ACP (Agent Client Protocol) agents to the application. It adds a new `/acp` slash command, a curated agent catalog, and the core data structures and interfaces needed to detect, launch, and interact with ACP-compatible agent CLIs. The changes are well-tested and modular, laying the groundwork for integrating multiple coding agents via a unified protocol.

The most important changes are:

**ACP Command and Integration:**
* Added the `/acp` slash command to the built-in commands list and registered it in the command handler dispatch table, enabling users to manage ACP agents and run turns against them. [[1]](diffhunk://#diff-fef46a04918a7d70ce73fa38397a982c8b7de7467edf3b886b19cd9188b028f6R440) [[2]](diffhunk://#diff-a8b5cdc3f81fe471935eee17f298f24f8566719aba1cbdb80f7e333df0b9fcf3R342)
* Added a test to ensure `/acp` is registered and appears in the correct order with other slash commands.

**Core ACP Agent Support:**
* Introduced `acp_client/active_turn.zig`, defining the `ActiveTurn` struct for managing the state and communication of an ACP agent turn, including update streaming, cancellation, and result publication. Includes thorough tests for thread safety and state transitions.
* Added `acp_client/agent_catalog.zig`, providing a curated catalog of popular ACP agents, detection logic for local and npx-installed agents, and utility functions for agent launch argument composition. Includes comprehensive tests for detection and catalog integrity.
* Implemented `acp_client/command_provider.zig`, defining the callback interfaces, request/result structures, and provider contract for integrating ACP command handling into the host application. Includes a test for contract correctness and memory safety.

**Wiring and Imports:**
* Wired up the ACP turn runner module in the main app commands file to enable ACP command handling.

These changes collectively enable the application to discover, launch, and interact with a variety of coding agents via the ACP protocol, setting the stage for advanced agent integration and management features.

### CODE REVIEW

File: src/core/app/app_commands.zig [hunk 1]
```diff
@@ -1106,6 +1152,178 @@ pub fn Handlers(comptime App: type) type {
...
+        /// Drains one tick of the active ACP turn: flushes streamed updates,
+        /// publishes the result when the turn finishes, and frees the turn.
+        pub fn drainAcpTurn(app: *App) !void {
+            const turn = app.acp_turn orelse return;
+            if (try turn.takeUpdate(app.alloc)) |text| {
+                defer app.alloc.free(text);
+                try app.pacer.enqueue(app.alloc, text);
+                app.shell.render_requests.request(.transcript);
+            }
+            if (turn.finished) {
```
Comment (DarlingtonDeveloper) [inline src/core/app/app_commands.zig:1289]: **[P1] Read the finished payload under the mutex**

`markFinished` publishes the result, flags, model arrays, and `finished` under `turn.mutex`, but this path reads them without that lock or an atomic publication primitive. This is a data race and can observe `finished` without a synchronized payload. Expose a locked finished-result snapshot or equivalent acquire/release publication.

File: src/core/acp_client/turn_runner.zig [hunk 2]
```diff
@@ -0,0 +1,637 @@
...
+    }
+    job.host.cwd = try alloc.dupe(u8, host.cwd);
+    const thread = std.Thread.spawn(.{}, Job.run, .{job}) catch |err| {
+        freeAgentConfig(alloc, job.agent_config);
+        alloc.free(job.config_path);
+        alloc.free(job.host.cwd);
+        alloc.destroy(job);
+        return err;
+    };
+    thread.detach();
```
Comment (DarlingtonDeveloper) [inline src/core/acp_client/turn_runner.zig:300]: **[P1] Join the detached warmup before destroying App**

The detached job retains pointers to the App-owned `SessionStore` and `Host.ctx`, but shutdown does not track or join it. Skipping session destruction while busy cannot keep the stack field or App alive. Exiting during an npx warmup leaves the worker using dangling pointers. Track and cancel/join the warmup, or heap-own its shared state.

**Review by DarlingtonDeveloper — CHANGES_REQUESTED** [review]
